ext/magic/ruby-magic.c

```diff
@@ -294,15 +294,19 @@ rb_mgc_get_paths(VALUE object)
 
 	MAGIC_CHECK_OPEN(object);
 
+  /* always prefer the value cached in @paths */
 	value = rb_ivar_get(object, id_at_paths);
-	if (!NIL_P(value) && !RARRAY_EMPTY_P(value) && !getenv("MAGIC"))
+	if (!NIL_P(value) && !RARRAY_EMPTY_P(value)) {
 		return value;
-
-	cstring = magic_getpath_wrapper();
-	value = magic_split(CSTR2RVAL(cstring), CSTR2RVAL(":"));
-	RB_GC_GUARD(value);
-
-	return value;
+  }
```

Comment:
Nit: Should this `}` be indented here?

Reply:
tabs vs spaces! :shakes-fist-at-editor: Will fix.